controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    position_controller:
      type: position_controllers/JointGroupPositionController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    effort_controller:
      type: effort_controllers/JointGroupEffortController
    
velocity_controller:
  ros__parameters:
    joints:
      - joint1
      - joint2
      - joint3
    
position_controller:
  ros__parameters:
    joints:
      - joint1
      - joint2
      - joint3

joint_trajectory_controller:
  ros__parameters:
    joints:
      - joint1
      - joint2
      - joint3
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

effort_controller:
  ros__parameters:
    joints:
      - joint1
      - joint2
      - joint3

    command_interfaces:
      - effort
    state_interfaces:
      - position
      - velocity
      - effort
